### PR TITLE
docs: add quick-start commands for testing in tests/README.md

### PR DESCRIPTION
**Pull request by retr0-kernel**
# Fixes

Resolved issue number : #1130

#### Describe the changes you have made in this PR -

Added a table with test commands in `tests/README.md` with three columns `Goal`, `Command`, and `When to use it`. The table includes commands for running different types of tests, such as unit tests with coverage, integration verification, live RCA end-to-end tests, single RCA fixtures, full suite including e2e, and synthetic scenarios. Each command is accompanied by a brief description of when to use it.

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

NOTE → Used AI to rummage through the existing `tests/README.md` and extract relevant information to create a concise table of test commands. The AI helped in structuring the information in a clear and organized manner.

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added (Not needed for documentation updates)
- [x] I have tested edge cases and understand how the code handles them (Not needed for documentation updates)
- [x] I have modified the AI output to follow this project's coding standards and conventions

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added** (Not needed for documentation updates)
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests (Not needed for documentation updates)
- [x] My code follows the project's style guidelines and conventions

---

### CODE REVIEW

**Comment by muddlebee** [comment]
@retr0-kernel take a look at review comments

File: tests/README.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Tests Catalog Naming Conventions
 
+## Quick-start commands
+
+| Goal | Command | When to use it |
+|---|---|---|
+| Run the default unit suite with coverage | `make test-cov` | First thing to run locally; no live infrastructure required. |
```
Comment (muddlebee) [inline tests/README.md:7]: test-cov runs pytest with coverage over the whole default tree minus synthetic ignores and -m "not synthetic"; it is not limited to unit tests. Prefer wording like default CI-aligned suite or standard pytest suite with coverage.

File: tests/README.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+| Verify all integration configs and clients | `make verify-integrations` | After adding or changing an integration. |
+| Run a live RCA end-to-end test | `make test-rca` | When you need to validate a full investigation against real services. |
+| Run a single RCA fixture | `make test-rca FILE=<name>` | When iterating on one specific alert scenario. |
+| Run the full suite including e2e | `make test-full` | Pre-release or CI; requires live infrastructure. |
+| Run synthetic scenarios (no live infra) | `make test-synthetic` | When testing scenario logic without external service dependencies. |
```
Comment (muddlebee) [inline tests/README.md:12]: Synthetic tests are LLM-driven RCA scenarios per repo docs; they still need configured LLM/API access even when customer infra is absent. Drop external service dependencies or qualify it as no customer live infra.

**Review by VaibhavUpreti — APPROVED** [review]
Thanks a lot for fixing this @retr0-kernel , and for the review @muddlebee ! Merging this for velocity, will fix in a separate patch
